### ISSUE TITLE
Add rule for redundant string enum values

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#946]
```swift
enum Numbers: String {
    case one = "one"
    case two = "two"
    case three = "three"
}
```

is redundant. Could be simply

```swift
enum Numbers: String {
    case one
    case two
    case three
}
```

**Comment by marcelofabri** [comment]
Should we handle `Int` enums as well? 🤔

**Comment by jpsim** [comment]
Maybe. That's not as clear to me.

**Comment by marcelofabri** [comment]
I'm implementing this and I'm uncertain if we should trigger only if all cases are redundant or if just one case (one violation per case).

This would trigger with the second approach, but not with the first:

```swift
private enum ConfigurationKey: String {
    case disabledRules = "disabled_rules"
    case enabledRules = "enabled_rules" // deprecated in favor of OptInRules
    case excluded = "excluded"
    case included = "included"
    case optInRules = "opt_in_rules"
    case reporter = "reporter"
    case useNestedConfigs = "use_nested_configs" // deprecated
    case whitelistRules = "whitelist_rules"
    case warningThreshold = "warning_threshold"
}
```
